pages/gestione/token/save.inc.php

```diff
@@ -0,0 +1,57 @@
+<?php
+// Gestione operazioni Token di Iscrizione
+
+// Controllo permessi amministratore
+if($_SESSION['permessi'] < SUPERUSER)  {
+    echo '<div class="error">Accesso negato. Permessi insufficienti.</div>';
+    return;
+}
+
+// Gestione operazioni
+if(isset($_REQUEST['op'])) {
+    switch($_REQUEST['op']) {
+        case 'genera_token':
+            $giorni_scadenza = intval($_REQUEST['giorni_scadenza']);
+            if($giorni_scadenza < 1 || $giorni_scadenza > 365) {
+                $giorni_scadenza = 30;
+            }
+            
+            // Genera token alfanumerico univoco
+            do {
+                $token = '';
+                $chars = 'ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789';
+                for($i = 0; $i < 16; $i++) {
+                    $token .= $chars[rand(0, strlen($chars) - 1)];
+                }
```

Comment:
Piccolo suggerimento: php dispone della funzione [random_bytes()](https://www.php.net/random_bytes) che è particolarmente appropriata per generare token casuali.
La funzione ritorna un output binario, ma tramite [bin2hex()](https://www.php.net/bin2hex) questo viene convertito in una stringa esadecimale.
```suggestion
                $token = bin2hex(random_bytes(8));
```